framework/src/mfem/bcs/MFEMBoundaryIntegratedBC.C

```diff
@@ -9,40 +9,39 @@
 
 #ifdef MFEM_ENABLED
 
-#include "MFEMScalarBoundaryIntegratedBC.h"
+#include "MFEMBoundaryIntegratedBC.h"
 
-registerMooseObject("MooseApp", MFEMScalarBoundaryIntegratedBC);
+registerMooseObject("MooseApp", MFEMBoundaryIntegratedBC);
 
 InputParameters
-MFEMScalarBoundaryIntegratedBC::validParams()
+MFEMBoundaryIntegratedBC::validParams()
 {
   InputParameters params = MFEMIntegratedBC::validParams();
   params.addClassDescription("Adds the boundary integrator to an MFEM problem for the linear form "
                              "$(f, v)_\\Omega$ "
                              "arising from the weak form of the forcing term $f$.");
-  params.addRequiredParam<MFEMScalarCoefficientName>(
-      "coefficient", "The coefficient which will be used in the integrated BC");
+  params.addParam<MFEMScalarCoefficientName>(
+      "coefficient", "1.", "The coefficient which will be used in the integrated BC");
   return params;
 }
 
-MFEMScalarBoundaryIntegratedBC::MFEMScalarBoundaryIntegratedBC(const InputParameters & parameters)
+MFEMBoundaryIntegratedBC::MFEMBoundaryIntegratedBC(const InputParameters & parameters)
   : MFEMIntegratedBC(parameters),
-    _coef_name(getParam<MFEMScalarCoefficientName>("coefficient")),
-    _coef(getScalarCoefficient(_coef_name))
```

Comment:
This is an aside - but is there a reason the `getScalarCoefficient` method doesn't behave like the majority of our interfaces?

Most `getXXX` methods in interfaces act on a parameter name, instead of a proper name. They query the parameter for you. Then we have `getXXXByName` if you want to specify the name directly.

Reply:
Are you suggesting that `getScalarCoefficient` should be called `getScalarCoefficientByName`?

Reply:
Yeah, that's what I think. and `getScalarCoefficient("coefficient")` should be the use case here.

Reply:
I'll add that to my list. :)